microsetta_interface/tests/test_integration.py

```diff
@@ -0,0 +1,161 @@
+        return encoded
+
+
+# we cannot have fine grain control over the state of the private-api
+# during integration testing. we also do not have control over the
+# order in which tests are executed. as such, treat each user with
+# care, and define new users on an as need basis.
+USER_NEW_EMAIL = 'user_new@foo.bar'
+USER_NEW = _fake_jwt(USER_NEW_EMAIL, True)
+
+USER_NEW_UNVERIFIED_EMAIL = 'user_unverified@foo.bar'
+USER_NEW_UNVERIFIED = _fake_jwt(USER_NEW_UNVERIFIED_EMAIL, False)
+
+USER_WITH_VALID_SAMPLE_EMAIL = "th-dq)wort@3'rey.i3l"
+USER_WITH_VALID_SAMPLE = _fake_jwt(USER_WITH_VALID_SAMPLE_EMAIL, True)
+
+
+@unittest.skipIf(not PRIVATE_API_AVAILABLE,
+                 "An instance of microsetta-private-api is not available")
+class IntegrationTests(unittest.TestCase):
+    def setUp(self):
+        self.patch_for_session = patch("microsetta_interface."
+                                       "implementation.session",
+                                       dict())
+        self.mock_session = self.patch_for_session.start()
+
+        app.app.testing = True
+        app.app.config['SECRET_KEY'] = 'secretsecret'
+        self.app = app.app.test_client()
+        self.app.__enter__()
+
+    def tearDown(self):
+        self.patch_for_session.stop()
+        self._logout()
+        self.app.__exit__(None, None, None)
+
+    def _html_page(self, resp):
+        return resp.get_data(as_text=True)
+
+    def _login(self, token):
+        self.mock_session[TOKEN_KEY_NAME] = token
+
+    def _logout(self):
+        try:
+            self.mock_session.pop(TOKEN_KEY_NAME)
+        except KeyError:
+            pass
+
+    def test_home(self):
+        # verify we access
+        resp = self.app.get('/home')
+        data = self._html_page(resp)
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('Sign Up', data)
+
+    def test_home_es_mx(self):
+        # make sure things come through in spanish
+        resp = self.app.get('/home', headers={'Accept-Language': 'es_MX'})
+        data = self._html_page(resp)
+        self.assertEqual(resp.status_code, 200)
+        self.assertNotIn('Sign Up', data)
+        self.assertIn('Registrarse', data)
+
+    def test_new_user(self):
+        # we should get /home for login if we are not already logged in
+        resp = self.app.get('/home')
+        self.assertEqual(resp.status_code, 200)
+
+        self._login(USER_NEW)
+
+        # once logged in, we should redirect
+        resp = self.app.get('/home')
+        self.assertEqual(resp.status_code, 302)
+
+        # this is a new user, so they should go to the account creation page
+        new_url = resp.headers['Location']
+        self.assertTrue(new_url.endswith('create_account'))
+
+        # make sure the creation page loads
+        resp = self.app.get(new_url)
+        self.assertEqual(resp.status_code, 200)
+        data = self._html_page(resp)
+        self.assertIn('<title>Microsetta Account Details</title>', data)
```

Comment:
Changing the page title in private api is going to mysteriously break microsetta interface builds the next time someone tries to put in a PR, that is going to be extremely confusing for someone.

Reply:
Titles come from -interface not -private-api. 

On the other hand, titles don't change frequently and anything we test in the content of the page is liable to change anyway?

Reply:
Ahh, as long as interface is testing its own behavior that's fine I guess.

Reply:
yup yup, thanks!